backend/app/modules/octant_rewards/general/service/finalized.py

```diff
@@ -15,6 +16,27 @@ def get_octant_rewards(self, context: Context) -> OctantRewardsDTO:
             context.epoch_details.epoch_num
         )
 
+        leftover_payload = LeftoverPayload(
+            staking_proceeds=int(pending_snapshot.eth_proceeds),
+            operational_cost=int(pending_snapshot.operational_cost),
+            community_fund=pending_snapshot.validated_community_fund,
+            ppf=pending_snapshot.validated_ppf,
+            total_withdrawals=int(finalized_snapshot.total_withdrawals),
+        )
+
+        unused_matched_rewards = context.epoch_settings.octant_rewards.leftover.extract_unused_matched_rewards(
+            int(finalized_snapshot.leftover), leftover_payload
+        )
+
+        allocations_sum = database.allocations.get_alloc_sum_by_epoch(
+            context.epoch_details.epoch_num
+        )
+        donated_to_projects = (
+            int(finalized_snapshot.matched_rewards)
+            - unused_matched_rewards
+            + allocations_sum
```

Comment:
In FE it was:
```
  const donatedToProjects = epochInfo
    ? epochInfo.matchedRewards +
      (totalUserDonationsWithPatronRewards - epochInfo.patronsRewards) -
      ethBelowThreshold
    : BigInt(0);
 ```
 Questions:
1. Why BE uses here `finalized_snapshot`? Value here should be updated during the AW too, depending on changing MR (`patron_rewards`) chunk.
2. Sanity check: `matched_rewards` here take `patron_rewards` into account?

Reply:
1. We need to do that both for pending snapshots and finalized ones (since in the finalized one there's some probability that whole MR won't be distributed between projects; not likely to happen but we need to cover it in the code too). Logic for the pending snapshots is here: backend/app/modules/octant_rewards/general/service/pending.py though.
2. Yes, MR include patron_rewards too.

Reply:
FE wrongly assumed that the entire MR went to projects even in a situation where no allocations were actually made. Possibly new BE logic is correct.